openhtmltopdf-pdfbox/src/main/java/com/openhtmltopdf/pdfboxout/PdfBoxFastLinkManager.java

```diff
@@ -224,23 +227,57 @@ private void addUriAsLink(RenderingContext c, Box box, PDPage page, float pageHe
 
 				PDAnnotationLink annot = new PDAnnotationLink();
 				annot.setAction(action);
-				if (!placeAnnotation(transform, linkShape, targetArea, annot))
+				if (!placeAnnotation(transform, linkShape, targetArea, new PDAnnotationLinkContainer(annot)))
 					return;
 
-				addLinkToPage(page, annot, box, target);
+				addLinkToPage(page, new PDAnnotationLinkContainer(annot), box, target);
 			} else {
 			    XRLog.general(Level.WARNING, "Could not find valid target for link. Link href = " + uri);
 			}
 		} else if (isURI(uri)) {
-			PDActionURI uriAct = new PDActionURI();
-			uriAct.setURI(uri);
 
 			Rectangle2D targetArea = checkLinkArea(page, c, box, pageHeight, transform, linkShape);
 			if (targetArea == null) {
 				return;
 			}
-			PDAnnotationLink annot = new PDAnnotationLink();
-			annot.setAction(uriAct);
+
+			PDAnnotationLink annotationLink = new PDAnnotationLink();
+			PDActionURI uriAct = new PDActionURI();
+			uriAct.setURI(uri);
+			annotationLink.setAction(uriAct);
+			PDAnnotationContainer annot = new PDAnnotationLinkContainer(annotationLink);
+
+			if ("true".equals(elem.getAttribute("data-embed-file"))) {
+				byte[] file = _sharedContext.getUserAgentCallback().getBinaryResource(uri);
+				if (file != null) {
+					try {
+						PDComplexFileSpecification fs = new PDComplexFileSpecification();
+						PDEmbeddedFile embeddedFile = new PDEmbeddedFile(_od.getWriter(), new ByteArrayInputStream(file));
+						embeddedFile.setSubtype(elem.getAttribute("data-content-type") != null ? elem.getAttribute("data-content-type") : "application/octet-stream");
```

Comment:
I think getattribute returns an empty string if not exists.

Reply:
fixed :) Thank you.